pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -666,9 +668,31 @@ func (s *DiscoveryServer) startPush(push *model.PushContext, full bool, edsUpdat
 	}
 	startTime := time.Now()
 	for _, p := range pending {
-		pushEvent := &PushEvent{edsUpdates, push, startTime, full}
-		s.pushQueue.Enqueue(p, pushEvent)
+		if proxyNeedsPush(p, req) {
+			s.pushQueue.Enqueue(p, &PushEvent{edsUpdates, push, startTime, req.Full})
+		}
+	}
+}
+
+func proxyNeedsPush(con *XdsConnection, req *model.UpdateReq) bool {
+	if !features.ScopePushes.Get() {
+		// If push scoping is not enabled, we push for all proxies
+		return true
 	}
+
+	// If no only namespaces specified, this request applies to all proxies
+	if len(req.TargetNamespaces) == 0 {
```

Comment:
this is never set from what I see..

Reply:
@howardjohn should this be set somewhere?

Reply:
It is set somewhere. https://github.com/istio/istio/pull/15995/files#diff-6713acfe7b9e389e7dcd429d08bee859R577 eds.go line 577. I thought I commented here, maybe was over slack

Reply:
Ok, so `TargetNamespaces` is always empty currently?

Reply:
No, it is set on eds.go line 577. 
```go
s.ConfigUpdate(model.UpdateRequest{Full: requireFull, TargetNamespaces: map[string]struct{}{namespace: {}}})
```